apps/server/src/modules/room/repo/entity/room.entity.ts

```diff
@@ -0,0 +1,40 @@
+import { Entity, Property } from '@mikro-orm/core';
+import { BaseEntityWithTimestamps } from '@shared/domain/entity/base.entity';
+import { Room } from '../../domain/do/room.do';
+
+export interface RoomEntityProps {
+	id?: string;
+	name: string;
+	color: string;
+	startDate?: Date;
+	untilDate?: Date;
+}
+
+@Entity({ tableName: 'rooms' })
+export class RoomEntity extends BaseEntityWithTimestamps {
+	@Property()
+	name: string;
+
+	@Property()
+	color: string;
+
+	@Property()
+	startDate?: Date;
+
+	@Property()
+	untilDate?: Date;
+
+	@Property({ persist: false })
+	domainObject: Room | undefined;
+
+	constructor(props: RoomEntityProps) {
+		super();
+		if (props.id) {
+			this.id = props.id;
+		}
+		this.name = props.name;
+		this.color = props.color;
+		this.startDate = props.startDate;
+		this.untilDate = props.untilDate;
+	}
```

Comment:
I think we should remove the constructor altogether. Domain objects should always have an id so the check is not necessary. Everything else is assigned when we call the default constructor from MikroORM `new RoomEntity(props)`.
Besides that the domain object has to take care of the state of the persistence entity. So we shouldn't have any logic in here.